### PR TITLE
update to typesript 2.x and vscode 1.x and extract changelog

### PR DESCRIPTION
**Pull request by 928PJY**
Update to typescript 2.x and VSCode 1.x.

### CODE REVIEW

File: src/VscPreviewExtension/package.json [hunk 1]
```diff
@@ -1,18 +1,18 @@
 {
   "name": "DfmPreview",
   "displayName": "DfmPreview",
-  "description": "Preview the DocFX Flavored Markdown and token tree in Visual Studio Code",
+  "description": "Preview the DocFX and token tree in Visual Studio Code",
```
Comment (hellosnow) [inline src/VscPreviewExtension/package.json]: >Preview the DocFX [](start = 18, length = 17)

how about "Preview DocFX generated html"?

**Comment by hellosnow** [comment]
:shipit:

File: src/VscPreviewExtension/package.json [hunk 2]
```diff
@@ -70,13 +70,16 @@
...
     "postinstall": "node ./node_modules/vscode/bin/install"
   },
   "devDependencies": {
-    "typescript": "^1.8.5",
-    "vscode": "^0.11.14"
+    "typescript": "^2.0.3",
+    "vscode": "^1.0.0",
+    "mocha": "^2.3.3",
+    "@types/node": "^6.0.40",
+    "@types/mocha": "^2.2.32"
```
Comment (qinezh) [inline src/VscPreviewExtension/package.json]: remove it if no need currently

File: src/VscPreviewExtension/package.json [hunk 3]
```diff
@@ -1,18 +1,18 @@
 {
   "name": "DfmPreview",
   "displayName": "DfmPreview",
-  "description": "Preview the DocFX Flavored Markdown and token tree in Visual Studio Code",
+  "description": "Preview DocFX generated Html and token tree in Visual Studio Code",
```
Comment (qinezh) [inline src/VscPreviewExtension/package.json]: Preview generated html and token tree by DocFX ?

Comment (928PJY) [inline src/VscPreviewExtension/package.json]: token tree is not generated by DocFX?

---
In reply to: [108832032](https://github.com/dotnet/docfx/pull/1507#discussion_r108832032) [](ancestors = 108832032)

File: src/VscPreviewExtension/package.json [hunk 4]
```diff
@@ -70,13 +70,16 @@
     ]
   },
   "scripts": {
-    "vscode:prepublish": "node ./node_modules/vscode/bin/compile",
-    "compile": "node ./node_modules/vscode/bin/compile -watch -p ./",
+    "vscode:prepublish": "tsc -p ./",
+    "compile": "tsc -watch -p ./",
```
Comment (qinezh) [inline src/VscPreviewExtension/package.json]: do we need depend on global command `tsc`? how about using `node ./node_modules/typescript/bin/tsc` directly

Comment (928PJY) [inline src/VscPreviewExtension/package.json]: VSCode has integrated with 'tsc', so exactly, we don't depend on global command 'tsc'.

---
In reply to: [108832323](https://github.com/dotnet/docfx/pull/1507#discussion_r108832323) [](ancestors = 108832323)

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by vicancy** [comment]
:shipit:

**Pull request merged**